abis_mapping/plugins/sites_geometry.py

```diff
@@ -0,0 +1,64 @@
+"""Provides extra frictionless geometry sites' data template checks for the package."""
+
+
+# Third-party
+import frictionless
+import frictionless.errors
+import attrs
+
+# Typing
+from typing import Iterator
+
+
+@attrs.define(kw_only=True, repr=False)
+class SitesGeometry(frictionless.Check):
+    """Checks whether geometry related fields are provided in a correct combination."""
+
+    # Check attributes
+    type = "sites-geometry"
+    Errors = [frictionless.errors.RowConstraintError]
+
+    # Occurrences site ids to be passed in from occurrence template.
+    occurrence_site_ids: set[str] = set()
+
+    def validate_row(self, row: frictionless.Row) -> Iterator[frictionless.Error]:
+        """Called to validate the given row (on every row).
+
+        Args:
+            row (frictionless.Row): The row to check the site's geometry for
+
+        Yields:
+            frictionless.Error: For when the check condition is violated.
+        """
+        # Extract values as bool
+        lat = row["decimalLatitude"] is not None
+        long = row["decimalLongitude"] is not None
+        datum = row["geodeticDatum"] is not None
+        wkt = row["footprintWKT"] is not None
+        site_id = row["siteID"] in self.occurrence_site_ids
+
+        # Perform check
+        if (
```

Comment:
Remove outer parentheses?

Reply:
i find it easier to read as a multi-line rather than a single-line. I put each AND group on a new line here. Lemme know if you would prefer the other way and I will remove the parentheses.